include/substrait-mlir/Dialect/Substrait/IR/SubstraitOps.td

```diff
@@ -415,6 +415,25 @@ def Substrait_EmitOp : Substrait_RelOp<"emit", [
   }];
 }
 
+def Substrait_FetchOp : Substrait_RelOp<"fetch", [
+    DeclareOpInterfaceMethods<InferTypeOpInterface>
+  ]> {
+  let summary = "Fetch operation";
+  let description = [{
+    Represents a `FetchOp` message together with the `RelCommon` and input
+    `Rel` message it contains. 
+  }];
+  let arguments = (ins
+    Substrait_Relation:$input,
+    I64Attr: $offset,
+    I64Attr: $count
```

Comment:
There are actually constraints for what values these numbers can take. From the spec:

| Property | Description | Required |
| ----- | ------------ | ----------- |
| Offset | A non-negative integer. Declares the offset for retrieval of records. | Optional, defaults to 0. |
| Count | A non-negative integer or -1. Declares the number of records that should be returned. -1 signals that ALL records should be returned. | Required |

We don't need to verify this in this PR but if we don't, then we should add a `TODO`.

The way to do this is with [attribute constraints](https://mlir.llvm.org/docs/DefiningDialects/Operations/#single-entity-constraint). `offset` probably just needs [`IntNonNegative`](https://github.com/llvm/llvm-project/blob/main/mlir/include/mlir/IR/CommonAttrConstraints.td#L780). For `count`, you'd also need to allow `-1`, which you could express as "(larger than -2) and (not equal to -1)", which I believe can be composed with existing tablegen constraints.

Reply:
It may also be worth it to make `offset` a default-valued attribute. This would make tablegen create C++ builders that omit that argument and set it to the default value (`0`).

Reply:
I think  "(larger than -2) and (not equal to -1)" is contradictory with wanting to allow -1? I think I could just do `ConfinedAttr<I64Attr, [IntMinValue<-1>]>`?